### DISCUSSION TITLE
can't load extension into sqlite studio

### DISCUSSION THREAD

**Discussion by CypherPotato** [pawelsalawa/sqlitestudio#4926] [General]
### Details
I'm trying to load [this extension](https://github.com/asg017/sqlite-xsv) into sqlite studio, but i'm getting an red exclamation mark telling that my file was not found.

![image](https://github.com/pawelsalawa/sqlitestudio/assets/17441745/dc245f63-ef4d-4838-b12c-af9e9b3b0b63)

But the file indeed is in the folder. 

![image](https://github.com/pawelsalawa/sqlitestudio/assets/17441745/37032539-5b07-466b-a66d-93fa5ad3d65a)

Also i can just type the path into the find and yea the file is there.

### Steps to reproduce
- Open SQLite Studio
- Add an database
- Go into extensions, add extension, navigate to file

### Operating system
Windows 11, x64

### SQLiteStudio version
3.4.4

**Comment by CypherPotato** [comment]
video of the issue: https://imgur.com/a/9bV6Wul

**Reply by Sbrjt** [reply]
Thnx for the video. I was wondering for a while how to load extension.

**Comment by pawelsalawa** [comment]
I'm not sure what is this `xcsvlib.dll`, but I've just downloaded file from their official release page:
![image](https://github.com/pawelsalawa/sqlitestudio/assets/481846/ed082a2f-16dd-47d8-80a8-5042478a52c9)
and after extracting I get the `xsv0.dll`, which I successfully loaded into SQLiteStudio:
![image](https://github.com/pawelsalawa/sqlitestudio/assets/481846/a98fe0ba-7179-4a52-8a7b-1fc98c38227f)
and even used it:
![image](https://github.com/pawelsalawa/sqlitestudio/assets/481846/2c1cfb50-b37d-483b-91cb-45bbd542de8c)

**Reply by CypherPotato** [reply]
`xsvlib` is just the same file, but in version v0.2.0, after I renamed it bc i thought it was an problem with the name. i've tried with `[sqlite-xsv-v0.2.1-alpha.11-loadable-windows-x86_64.zip](https://github.com/asg017/sqlite-xsv/releases/download/v0.2.1-alpha.11/sqlite-xsv-v0.2.1-alpha.11-loadable-windows-x86_64.zip)` but got no success. i'm uninstalling sqlite studio and installing it again to see if it fixes. also clearing the temp.

**Reply by CypherPotato** [reply]
update: after reinstalling it I got it working.

thank you.

**Reply by pawelsalawa** [reply]
The filename of extension is essential, because the initialization function is - by default - derived from its file name. If you change file name, then you can specify initialization function name by yourself and it will still work:
![image](https://github.com/pawelsalawa/sqlitestudio/assets/481846/3c5360c4-33c3-42f1-ba51-2149e5a331b0)

This is documented in SQLite's `load_extension()` C function: https://www.sqlite.org/c3ref/load_extension.html
